### PR TITLE
fix(content-uploader): use refs for items and itemIds

### PR DESCRIPTION
**Pull request by ahorowitz123**

### CODE REVIEW

File: src/elements/content-uploader/ContentUploader.tsx [hunk 1]
```diff
@@ -994,10 +990,13 @@ class ContentUploader extends Component<ContentUploaderProps, State> {
...
 
         if (items.length === 0) {
+            this.itemIdsRef.current = {};
             state.itemIds = {};
             state.errorCode = '';
         }
 
+        this.itemsRef.current = items;
+
         this.setState(state as Pick<State, keyof State>, callback);
```
Comment (bfoxx1906) [inline src/elements/content-uploader/ContentUploader.tsx:1000]: If we surrounded this call with flushSync would it solve a lot of our problems?

Comment (ahorowitz123) [reply src/elements/content-uploader/ContentUploader.tsx:1000]: no unfortunately. That was the first thing I tried

File: src/elements/content-uploader/ContentUploader.tsx [hunk 2]
```diff
@@ -289,9 +294,8 @@ class ContentUploader extends Component<ContentUploaderProps, State> {
      */
     getNewFiles = (files: Array<UploadFileWithAPIOptions | File>): Array<UploadFileWithAPIOptions | File> => {
         const { rootFolderId } = this.props;
-        const { itemIds } = this.state;
```
Comment (JChan106) [inline src/elements/content-uploader/ContentUploader.tsx:292]: is `itemIds` used anywhere now?

Comment (ahorowitz123) [reply src/elements/content-uploader/ContentUploader.tsx:292]: No its not. But to reduce risk I decided to leave it in for now. I can create a followup ticket to remove it

**Review by JChan106 — APPROVED** [review]

**Review by jmcbgaston — APPROVED** [review]
LGTM

**Review by patlm — APPROVED** [review]

**Pull request merged**